### PR TITLE
UX: Improve error messages for minimum and maximum username lengths.

### PR DESCRIPTION
**Pull request by nbianca**

### CODE REVIEW

File: lib/validators/max_username_length_validator.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+  def initialize(opts = {})
+    @opts = opts
+    @username = ""
+  end
+
+  def valid_value?(val)
+    return false if val < SiteSetting.min_username_length
+
+    users = User.where('length(username) > ?', val).limit(1)
```
Comment (ZogStriP) [inline lib/validators/max_username_length_validator.rb]: It's even faster to ask postgresql if a record exists?

```ruby
return true unless User.exists?("length(username) > ?", val)
```

File: lib/validators/max_username_length_validator.rb [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    return true if users.size == 0
+
+    @username = users[0].username
+    false
+  end
+
+  def error_message
+    if @username == ""
```
Comment (ZogStriP) [inline lib/validators/max_username_length_validator.rb]: `if @username.blank?` is more readable here ;)

File: lib/validators/max_username_length_validator.rb (hunk 1 above)
Comment (ZogStriP) [reply lib/validators/max_username_length_validator.rb]: Actually, nevermind. You use the username later on. Carry on :)

**Comment by tgxworld** [comment]
Merged in https://github.com/discourse/discourse/commit/e0d7cdac12a180697b3939967c8315184899bfa4. @nbianca There are some minor changes I made but nothing major. Nice work :+1:

**Pull request closed without merging**